src/matcha_ml/storage/azure_storage.py

```diff
@@ -189,6 +191,26 @@ def _get_blob_names(self, container_name: str) -> Set[str]:
         """
         return set(self._get_container_client(container_name).list_blob_names())
 
+    def get_hash_remote_state(self, container_name: str, blob_name: str) -> str:
+        """Get hash of remote matcha state file.
+
+        Args:
+            container_name (str): Azure storage container name
+            blob_name (str): blob name
+
+        Returns:
+            str: Hash contents of the blob in hexadecimal string
+        """
+        remote_hash = None
+        blob_client = self._get_blob_client(
+            container_name=container_name, blob_name=blob_name
+        )
+        with tempfile.NamedTemporaryFile() as tmp:
+            self.download_file(blob_client, tmp.name)
```

Comment:
Could we speed this up by only downloading and hashing the matcha.state file? If I've understood right, this is hashing the entirety of the blob.

Reply:
It downloads only the `matcha.state` file from remote storage.